### PR TITLE
Remove remaining appengine interfaces

### PR DESCRIPTION
**Pull request by loosebazooka**
~~blocked by #753~~

### CODE REVIEW

**Comment by loosebazooka** [comment]
I think we can ignore the code coverage on this, but this PR is ready to go.

File: src/main/java/com/google/cloud/tools/appengine/api/auth/Auth.java [hunk 1]
```diff
@@ -17,12 +17,73 @@
...
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
 
-public interface Auth {
-  void login(String user) throws AppEngineException;
+/** Interface to gcloud auth commands. */
+public class Auth {
+  private static final Pattern emailPattern =
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/appengine/api/auth/Auth.java]: unrelated nit: `Pattern` is immutable, so I always go for all caps.

File: src/main/java/com/google/cloud/tools/appengine/api/debug/GenRepoInfoFile.java [hunk 2]
```diff
@@ -17,9 +17,49 @@
...
+import com.google.cloud.tools.appengine.cloudsdk.internal.args.GcloudArgs;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessHandlerException;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 
-/** Main interface for generating source context files, needed by the Cloud Debugger. */
-public interface GenRepoInfoFile {
+/** Interface for generating source context files using gcloud, needed by the Cloud Debugger. */
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/appengine/api/debug/GenRepoInfoFile.java]: Do you want to use the term "Interface"?

File: src/main/java/com/google/cloud/tools/appengine/api/versions/Versions.java [hunk 3]
```diff
@@ -14,25 +14,25 @@
...
 import com.google.cloud.tools.appengine.cloudsdk.internal.args.GcloudArgs;
 import com.google.cloud.tools.appengine.cloudsdk.process.ProcessHandlerException;
 import com.google.common.base.Preconditions;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
-/** Cloud SDK based implementation of {@link AppEngineVersions}. */
-public class CloudSdkAppEngineVersions implements AppEngineVersions {
+/** Interface with gcloud versions command. */
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/appengine/api/versions/Versions.java]: ditto?

File: src/main/java/com/google/cloud/tools/appengine/api/debug/GenRepoInfoFile.java (hunk 2 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/appengine/api/debug/GenRepoInfoFile.java]: I did, but I don't think my choice of words was great, will change it.

**Pull request merged**